src/components/Tables/Table.tsx

```diff
@@ -0,0 +1,100 @@
+"use client";
+
+import React, { useState, useEffect } from "react";
+import DataTable, { TableColumn } from "react-data-table-component";
+import TableFilterBar, { FilterText } from "./TableFilterBar";
+
+interface Row {
+    [key: string]: string | number;
+}
+interface Headers {
+    [key: string]: string;
+}
+
+export interface Datum {
+    [key: string]: string | number | null;
+}
+
+interface Props {
+    data: Datum[];
+    headers: Headers;
+}
+
+const itemIncludesFilterText = (headers: Headers, item: Datum, filterText: FilterText): boolean => {
+    for (const key of Object.keys(headers)) {
+        if (
+            !(item[key] ?? "")
+                .toString()
+                .toLowerCase()
+                .includes((filterText[key] ?? "").toLowerCase())
+        ) {
+            return false;
+        }
+    }
+    return true;
+};
+
+const filteredNoNullItems = (headers: Headers, data: Datum[], filterText: FilterText): Row[] => {
+    return data.reduce((filteredNoNullItems: Row[], item: Datum) => {
```

Comment:
can the loop variable not be the same name as the function please?

Reply:
refactored as suggested